design/variable_command_design.md

```diff
@@ -0,0 +1,181 @@
+
+```bash
+$ conjur-cli variable get secrets/test_secret foo/bar
+
+## Current
+requests.exceptions.HTTPError: 404 Client Error: Not Found for url: https://localhost/secrets/myorg/variable/secrets%2Ftest_secret%2Cfoot%2Fbar
+
+## Should be
+Error: Data either not found or variable value does not exist
+
+<Variable command help screen>
+```
+
+*Set secrets*
+
+```bash
+$ conjur-cli variable set foo/bar mysecret
+
+## Current
+Value set: 'foo/bar'
+
+## Should be
+Success! Data written to: 'foo/bar'
+```
+
+*Get secrets*
+
+```bash
+$ conjur-cli variable get foo/bar
+mysecret%
+```
+
+*Get a secret without inputting which secret to get*
+
+```bash
+$ conjur-cli variable get __
+
+## Current 
+usage: conjur-cli variable get [-h] variable_id [variable_id ...]
+
+## Should be
+Error: Command 'variable' requires the following arguments: variable_id
+
+<Variable command help screen>
+```
+
+*Get a secret without inputting which variable_id / its value*
+
+```bash
+$ conjur-cli variable set __ __
+
+## Current
+usage: conjur-cli variable set [-h] variable_id value
+conjur-cli variable set: error: the following arguments are required: variable_id, value
+
+## Should be
+Error: Command 'variable' requires the following arguments: variable_id value
+
+<Variable command help screen>
+```
+
+*Get secret without inputting the variable_id’s value*
+
+```bash
+$ conjur-cli variable set foo/bar __
+
+## Current
+usage: conjur-cli variable set [-h] variable_id value
+conjur-cli variable set: error: the following arguments are required: value
+
+## Should be
+Error: Command 'variable' requires the following arguments: variable_id value
+
+<Variable command help screen>
+```
+
+*Get secrets with more inputs than excepted*
+
+```bash
+$ conjur-cli variable set foo/bar mysecret sfdfs
+
+## Current
+usage: conjur-cli [-h] [-v] [-l URL] [-a ACCOUNT] [-c CA_BUNDLE] [--insecure] [-u LOGIN_ID] [-k API_KEY] [-p PASSWORD] [-d]
+         [--verbose]
+         {list,variable,policy} ...
+conjur-cli: error: unrecognized arguments: sfdfs
+
+## Should be:
+Should the last input be ignored? Should it take only the arguments it needs (first 2) as we do in Ruby CLI?
+```
+
+*Edge case: spaces in variable_ids*
+
+```bash
+conjur-cli variable get "secrets/test with space" foo/bar
+{
+    "foo/bar": "mysecret",
+    "secrets/test with space": "mysecrettwo"
+}
+```
+
+Notice the `""` that needs to surround variable_ids that have spaces in them to work properly
+
+
+
+### Open questions
+
+1. If extra arguments are provided, what should be the experience? See second to last flow.
+
+2. Should we return error codes? 404 if not found on secrets?
+
+### Delivery plan
+
+1. Ensure UTs and integration tests exist and are GA quality (3 days)
+   1. Sit with Elad to map out current tests and what is missing
+   2. Implement missing tests (if required)
+2. Improve error handling experience (3 days)
+   1. ***Blocker*** - need to get full UX outlines from Sharon + Inbal
```

Comment:
I think we agreed on \" and spaces
conjur variable get "var1" "var2"
OK?
Ill update the Epic also

Reply:
yes but I am talking about the actual help screen, info, and error statements here